system-dependencies.sh

```diff
@@ -999,6 +1011,41 @@ function check_simulators ()
 	fi
 }
 
+function check_dotnet ()
+{
+	if test -n "$IGNORE_DOTNET"; then return; fi
+
+	local DOTNET_VERSION
+	local DOTNET_INSTALL_DIR
+	local DOTNET_URL
+	local DOTNET_FILENAME
+
+	DOTNET_VERSION=$(grep ^DOTNET_VERSION= Make.config | sed 's/.*=//')
+	DOTNET_URL=$(grep ^DOTNET_URL= Make.config | sed 's/.*=//')
+	DOTNET_INSTALL_DIR=/usr/local/share/dotnet/sdk/"$DOTNET_VERSION"
+	DOTNET_FILENAME=$(basename "$DOTNET_URL")
+
+	if test -d "$DOTNET_INSTALL_DIR"; then
+		ok "Found dotnet $DOTNET_VERSION (exactly $DOTNET_VERSION is required)."
+		return
+	fi
+
+	if test -z "$PROVISION_DOTNET"; then
+		fail "You must install dotnet $DOTNET_VERSION. You can download it from ${COLOR_BLUE}$DOTNET_URL${COLOR_RESET}."
+		fail "Alternatively you can ${COLOR_MAGENTA}export IGNORE_DOTNET=1${COLOR_RED} to skip this check."
```

Comment:
should this include a message about the `--ignore-dotnet` option?

Reply:
You'll usually see this message after running `make`, and then it's must harder to pass the `--ignore-dotnet` option to this script (you have to edit the makefiles) than to export an environment variable.